### PR TITLE
Adds rollerskates, wheely can do tricks

### PR DESCRIPTION
**Pull request by Mickyan**
This turns wheelys into a skateboard subtype so they are no longer a copy paste of old skateboard code. They have lower stability compared to skateboards.
I've also made a whole bunch of small adjustments in the process to make skateboard code cleaner and more modular, such as turning the improvised skateboard to its own subtype so that construction steps don't need to be overwritten for all subtypes

Adds two subtypes of wheelys:
-Rollerskates: more stable than wheelys but slow you down when the wheels are retracted
-Skishoes: For skiing! Only work on snow, obviously

I wanted to add some fun alternatives to skateboards with their own drawbacks, and the ski shoes were an opportunity to do something fun for icemoon maps

Screenshot_5
Screenshot_6

:cl:
add: Added rollerskates and skishoes, available at the library's fun vendor
code: wheelys are now a skateboard subtype and more modular
code: improvised skateboard is now it's own subtype
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Does this mean that wheelies will no longer let you Z O O M around at super-speed?

Will ski shoes affect the balance of icebox megafauna?

**Comment by EOBGames** [comment]
> Does this mean that wheelies will no longer let you Z O O M around at super-speed?
> 
> Will ski shoes affect the balance of icebox megafauna?

given that icemoon snow deliberately has no slowdown, that doesn't seem likely

**Review by Twaticus — APPROVED** [review]

File: code/modules/vehicles/vehicle_actions.dm [hunk 1]
```diff
@@ -233,6 +233,6 @@
 			V.pass_flags &= ~PASSTABLE
 		if(locate(/obj/structure/table) in V.loc.contents)
 			V.grinding = TRUE
-			V.icon_state = "[V.board_icon]-grind"
+			V.icon_state = "[initial(V.icon_state)]-grind"
```
Comment (Timberpoes) [inline code/modules/vehicles/vehicle_actions.dm:236]: Not gonna consider this one a merge blocker, but if you have the time I'd love to see all the Vs and Ls replaced as above.

File: code/modules/vehicles/scooter.dm [hunk 2]
```diff
@@ -256,47 +250,61 @@
...
+	name = "ski shoes"
+	desc = "Uses patented retractable wheel technology. Never sacrifice speed for style - not that this provides much of either."
+	instability = 8
+	wheel_name = "skis"
+
+/obj/vehicle/ridden/scooter/skateboard/wheelys/skishoes/Initialize()
+	. = ..()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.vehicle_move_delay = 1
+	D.allowed_turf_typecache = typecacheof(/turf/open/floor/plating/asteroid/snow/icemoon)
```
Comment (Timberpoes) [inline code/modules/vehicles/scooter.dm:310]: Ditto.

File: code/modules/vehicles/scooter.dm [hunk 3]
```diff
@@ -256,47 +250,61 @@
...
-		H.Paralyze(30)
-		H.adjustStaminaLoss(10)
-		var/head_slot = H.get_item_by_slot(ITEM_SLOT_HEAD)
-		if(!head_slot || !(istype(head_slot,/obj/item/clothing/head/helmet) || istype(head_slot,/obj/item/clothing/head/hardhat)))
-			H.adjustOrganLoss(ORGAN_SLOT_BRAIN, 1)
-			H.updatehealth()
-		visible_message("<span class='danger'>[src] crashes into [A], sending [H] flying!</span>")
-		playsound(src, 'sound/effects/bang.ogg', 50, TRUE)
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.vehicle_move_delay = 1.5
```
Comment (Timberpoes) [inline code/modules/vehicles/scooter.dm:298]: Ditto.

File: code/modules/vehicles/scooter.dm [hunk 4]
```diff
@@ -256,47 +250,61 @@
...
-/obj/vehicle/ridden/scooter/wheelys/Initialize()
+/obj/vehicle/ridden/scooter/skateboard/wheelys/Initialize()
 	. = ..()
 	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
 	D.vehicle_move_delay = 0
-	D.set_vehicle_dir_layer(SOUTH, ABOVE_MOB_LAYER)
-	D.set_vehicle_dir_layer(NORTH, OBJ_LAYER)
-	D.set_vehicle_dir_layer(EAST, OBJ_LAYER)
-	D.set_vehicle_dir_layer(WEST, OBJ_LAYER)
+	D.set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0), TEXT_SOUTH = list(0), TEXT_EAST = list(0), TEXT_WEST = list(0)))
```
Comment (Timberpoes) [inline code/modules/vehicles/scooter.dm:269]: Are you able to rename this from a single letter var?

**Review by Timberpoes — CHANGES_REQUESTED** [review]

**Comment by Timberpoes** [comment]
Love what you've done here. Twaticus have give you the spriting badge of approval, so once the niggly bits of code review are addressed I'd be happy to approve this with a view to merging.

I'm mindful that there *may* be another Icebox related merge with minor mapping changes, but their review is likely to take longer than yours so you'll probably get to win the merge lottery for avoiding any potential mapping conflicts.

File: code/modules/vehicles/vehicle_actions.dm (hunk 1 above)
Comment (Mickyan) [reply code/modules/vehicles/vehicle_actions.dm:236]: might as well while I'm here

**Review by Timberpoes — APPROVED** [review]
Let's get readyyyyy to rrrumbleeeee.

Looking good from me. It's been up 4 days with nary a peep outta anyone, so let's get this puppy merged and see what breaks!

**Pull request merged**